builtInFunctions/multiESDTNFTTransfer.go

```diff
@@ -0,0 +1,452 @@
+		listTokenID[i] = vmInput.Arguments[tokenStartIndex]
+		nonce := big.NewInt(0).SetBytes(vmInput.Arguments[tokenStartIndex+1]).Uint64()
+		quantityToTransfer := big.NewInt(0).SetBytes(vmInput.Arguments[tokenStartIndex+2])
+		listEsdtData[i], err = e.transferOneTokenOnSenderShard(acntSnd, acntDst, dstAddress, listTokenID[i], nonce, quantityToTransfer, verifyPayable)
+		if err != nil {
+			return nil, err
+		}
+
+		logEntry := newEntryForNFT(vmcommon.BuiltInFunctionMultiESDTNFTTransfer, vmInput.CallerAddr, listTokenID[i], nonce)
+		logEntry.Topics = append(logEntry.Topics, acntDst.AddressBytes())
+		vmOutput.Logs[i] = logEntry
+	}
+
+	err = e.createESDTNFTOutputTransfers(vmInput, vmOutput, listEsdtData, listTokenID, dstAddress)
+	if err != nil {
+		return nil, err
+	}
+
+	return vmOutput, nil
+}
+
+func (e *esdtNFTMultiTransfer) transferOneTokenOnSenderShard(
+	acntSnd vmcommon.UserAccountHandler,
+	acntDst vmcommon.UserAccountHandler,
+	dstAddress []byte,
+	tokenID []byte,
+	nonce uint64,
+	quantityToTransfer *big.Int,
+	verifyPayable bool,
+) (*esdt.ESDigitalToken, error) {
+	if quantityToTransfer.Cmp(zero) <= 0 {
+		return nil, ErrInvalidNFTQuantity
+	}
+
+	esdtTokenKey := append(e.keyPrefix, tokenID...)
+	esdtData, err := getESDTNFTTokenOnSender(acntSnd, esdtTokenKey, nonce, e.marshalizer)
+	if err != nil {
+		return nil, err
+	}
+
+	if esdtData.Value.Cmp(quantityToTransfer) < 0 {
+		return nil, ErrInvalidNFTQuantity
+	}
+	esdtData.Value.Sub(esdtData.Value, quantityToTransfer)
+
+	_, err = saveESDTNFTToken(acntSnd, esdtTokenKey, esdtData, e.marshalizer, e.pauseHandler)
+	if err != nil {
+		return nil, err
+	}
+
+	esdtData.Value.Set(quantityToTransfer)
+
+	if !check.IfNil(acntDst) {
+		err = e.addNFTToDestination(dstAddress, acntDst, esdtData, esdtTokenKey, verifyPayable)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	return esdtData, nil
+}
+
+func (e *esdtNFTMultiTransfer) loadAccountIfInShard(dstAddress []byte) (vmcommon.UserAccountHandler, error) {
+	if e.shardCoordinator.SelfId() != e.shardCoordinator.ComputeId(dstAddress) {
+		return nil, nil
+	}
+
+	accountHandler, errLoad := e.accounts.LoadAccount(dstAddress)
+	if errLoad != nil {
+		return nil, errLoad
+	}
+	userAccount, ok := accountHandler.(vmcommon.UserAccountHandler)
+	if !ok {
+		return nil, ErrWrongTypeAssertion
+	}
+
+	return userAccount, nil
+}
+
+func (e *esdtNFTMultiTransfer) createESDTNFTOutputTransfers(
+	vmInput *vmcommon.ContractCallInput,
+	vmOutput *vmcommon.VMOutput,
+	listESDTTransferData []*esdt.ESDigitalToken,
+	listTokenIDs [][]byte,
+	dstAddress []byte,
+) error {
+
+	multiTransferCallArgs := make([][]byte, 0, argumentsPerTransfer*uint64(len(listESDTTransferData))+1)
+	numTokenTransfer := big.NewInt(int64(len(listESDTTransferData))).Bytes()
+	multiTransferCallArgs = append(multiTransferCallArgs, numTokenTransfer)
+
+	for i, esdtTransferData := range listESDTTransferData {
+
+		multiTransferCallArgs = append(multiTransferCallArgs, listTokenIDs[i])
```

Comment:
extract in a new function

Reply:
it is only a for and argument creation inside - it would require a lot of returns.